cpp/inc/bond/core/config.h

```diff
@@ -40,3 +40,9 @@
 #define BOND_CONSTEXPR          BOOST_CONSTEXPR
 #define BOND_CONSTEXPR_OR_CONST BOOST_CONSTEXPR_OR_CONST
 #define BOND_STATIC_CONSTEXPR   BOOST_STATIC_CONSTEXPR
+
+#ifdef BOND_HEADER_ONLY
```

Comment:
The default for this needs to be header-only, to avoid breaking people who aren't linking with libbond. Also, consider something along the lines of

```
#define BOND_LIB_TYPE_HEADER 1
#define BOND_LIB_TYPE_STATIC 2
// #define BOND_LIB_TYPE_DYNAMIC 3 // not yet implemented
// #define BOND_LIB_TYPE_DYNAMIC 4 // not yet implemented

#ifndef BOND_LIB_TYPE
#define BOND_LIB_TYPE BOND_LIB_TYPE_HEADER
#endif
```